web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte

```diff
@@ -139,6 +142,14 @@
         title="Download"
       />
     {/if}
+    {#if showShareButton}
```

Comment:
Can you restrict it to asset owners ?

Reply:
Initially I included restriction on owner but tested selecting multiple images from the grid view including those that were from partner share. The partner photos showed up in the share link as well. Should a user be able to share partner share photos?

Reply:
I think owner or partner can both share it

Reply:
Yep, but that's an issue for assets in shared albums
![image](https://github.com/immich-app/immich/assets/74269598/22b483d3-b220-48a3-b569-22b02a6ac3ae)

Reply:
I'll take a look at assets in shared albums

Reply:
I was able to reproduce by inviting a user to the shared album that was also _not_ a partner share. The same error is displayed if selecting the image from the main album page and selecting share as well. So seems like an existing but not necessarily related to this change.

Created https://github.com/immich-app/immich/issues/7604 for tracking